tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -453,27 +455,35 @@ func getMasterIpOnSite(ctx context.Context, client clientset.Interface, primaryS
 		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
 	}
 	framework.Logf("Site esx map : %v", siteEsxMap)
-	// Assuming atleast one master is on that site
 	vcAddress := e2eVSphere.Config.Global.VCenterHostname
-	for _, masterIp := range allMasterIps {
-		cmd := "export GOVC_INSECURE=1;"
-		cmd += fmt.Sprintf("export GOVC_URL='https://administrator@vsphere.local:Admin!23@%s';", vcAddress)
-		cmd += fmt.Sprintf("govc vm.info --vm.ip=%s;", masterIp)
-		result, err := sshExec(sshClientConfig, masterIp, cmd)
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
-		hostIp := strings.Split(result.Stdout, "Host:")
-		host := strings.TrimSpace(hostIp[1])
-		if siteEsxMap[host] {
-			masterIpOnSite = masterIp
-			break
+	// Assuming atleast one master is on that site
+	waitErr := wait.PollImmediate(healthStatusPollInterval, pollTimeout*2, func() (bool, error) {
+		for _, masterIp := range allMasterIps {
+			cmd := "export GOVC_INSECURE=1;"
+			cmd += fmt.Sprintf("export GOVC_URL='https://administrator@vsphere.local:Admin!23@%s';", vcAddress)
```

Comment:
get the user and password from e2e config, we may be using a different user while setting up the testbed (minimal permission user)